conf/utd_ganymede.config

```diff
@@ -0,0 +1,24 @@
+//Profile config names for nf-core/configs
+params {
+  config_profile_description = 'University of Texas at Dallas HPC cluster profile provided by nf-core/configs'
+  config_profile_contact = 'Edmund Miller(@emiller88)'
+  config_profile_url = 'http://docs.oithpc.utdallas.edu/'
+}
+
+singularity {
+  enabled = true
+  envWhitelist='SINGULARITY_BINDPATH'
+  autoMounts = true
+}
+
+process {
+  beforeScript = 'module load Singularity/2.4.5'
+  executor = 'slurm'
+  queue = 'genomics'
+}
+
+params {
+  max_memory = 32.GB
```

Comment:
Is this the total memory you have on each node (`--mem`) or the `--mem-per-cpu`? This value is there to reflect the former. Just thought Id ask because this setting implies you have `2.GB` per cpu on your cluster.

Reply:
1. How do I get/calculate the max_memory?

2. Side question for my learning, will nextflow use multiple nodes to compute a task? For example for STAR, each node has 32GB(which is slightly too little for a human genome), will it be able to pull in two nodes to have enough memory? I couldn't get it to work with Snakemake

Reply:
1. You would have to ask your cluster admins about this Im afraid. Some clusters will have nodes with exactly the same resources and others wont. Ideally, you would set these values to the minimum resources available on any given node in that queue. This just ensures that the pipelines wont break if you over-request resources by being allocated a smaller node at run-time.

2. You mean MPI jobs? As far as I know this isnt possible with Nextflow yet if using `slurm`. Each job will be submitted to an individual node.

Reply:
1. I know that each of our nodes have the same resources. I uses `scontrol show node <one-of-the-nodes>` to get
```
NodeName=compute-7-2-0 Arch=x86_64 CoresPerSocket=8
   CPUAlloc=16 CPUTot=16 CPULoad=16.04
   AvailableFeatures=(null)
   ActiveFeatures=(null)
   Gres=(null)
   NodeAddr=compute-7-2-0 NodeHostName=compute-7-2-0 Version=17.11
   OS=Linux 3.10.0-862.9.1.el7.x86_64 #1 SMP Mon Jul 16 16:29:36 UTC 2018
   RealMemory=32106 AllocMem=0 FreeMem=11664 Sockets=2 Boards=1
   State=ALLOCATED ThreadsPerCore=1 TmpDisk=0 Weight=1 Owner=N/A MCS_label=N/A
   Partitions=genomics
   BootTime=2019-10-25T14:03:02 SlurmdStartTime=2020-02-07T09:09:10
   CfgTRES=cpu=16,mem=32106M,billing=16
   AllocTRES=cpu=16,mem=32106M,billing=16
   CapWatts=n/a
   CurrentWatts=0 LowestJoules=0 ConsumedJoules=0
   ExtSensorsJoules=n/s ExtSensorsWatts=0 ExtSensorsTemp=n/s
```
So like I thought, there's 16 cores, and 32GB of RAM. So is that calculation correct for 2GB per CPU core. But you said "per cpu on your cluster" and there's two sockets per node, so two cpus. So it should be 16?

Reply:
I doesnt look like you have a max `--mem-per-cpu` set there. On our cluster we have `AllocMem=68480` and `AllocTRES=cpu=32,mem=68480M`. Maybe because the nodes have quite a small memory overall. I think the settings you have proposed are good 👍